### PR TITLE
Add PROXY protocol support

### PR DESCRIPTION
**Pull request by LXGaming**

### CODE REVIEW

File: proxy/src/main/java/com/velocitypowered/proxy/config/VelocityConfiguration.java [hunk 1]
```diff
@@ -41,6 +41,10 @@
     @ConfigKey("online-mode")
     private boolean onlineMode = true;
 
+    @Comment("Enables compatibility with the PROXY protocol.")
```
Comment (Leymooo) [inline proxy/src/main/java/com/velocitypowered/proxy/config/VelocityConfiguration.java]: PROXY -> HAPROXY

File: proxy/src/main/java/com/velocitypowered/proxy/config/VelocityConfiguration.java [hunk 2]
```diff
@@ -41,6 +41,10 @@
...
+    @ConfigKey("proxy-protocol")
+    private boolean proxyProtocol = false;
```
Comment (Leymooo) [inline proxy/src/main/java/com/velocitypowered/proxy/config/VelocityConfiguration.java]: move this to Advanced

File: proxy/src/main/java/com/velocitypowered/proxy/config/VelocityConfiguration.java (hunk 1 above)
Comment (electronicboy) [reply proxy/src/main/java/com/velocitypowered/proxy/config/VelocityConfiguration.java]: PROXY is correct here, given that it's talking about the actual protocol itself and not a specific implementation

**Comment by astei** [comment]
Going to give this a try today.

**Comment by astei** [comment]
Verified this works. Thanks for the contribution! :)

**Pull request merged**